### PR TITLE
Allow selection of which warnings to log

### LINKED ISSUE #11500: Warnings logging doesn't log warnings
### Description

The Astropy logging mechanism provides the `enable_warnings_logging()` function, which is claimed to enable logging of `warnings.warn()` messages. This does not appear to work.

### Expected behavior

Warnings signaled by `warnings.warn` should be recorded in the log.

### Actual behavior

Such warnings do not appear.

### Steps to Reproduce

```python
import warnings
from astropy import log
import astropy.logger
import logging

try:
    log.enable_warnings_logging()
except astropy.logger.LoggingError:
    pass

with log.log_to_file("test.log"):
    log.warning("logger warning")
    warnings.warn("warnings warning")
```

The file `test.log` should contain both "logger warning" and "warnings warning" but it only contains "logger warning".

(The exception handling is because the only way to ensure the astropy warnings logging mechanism is turned on is to call the function and catch the exception if it's already turned on; it is in my completely default setup.)

### System Details

```
>>> import platform; print(platform.platform())
Linux-3.10.0-862.14.4.el7.x86_64-x86_64-with-debian-buster-sid
>>> import sys; print("Python", sys.version)
Python 3.7.6 | packaged by conda-forge | (default, Jun  1 2020, 18:57:50)
[GCC 7.5.0]
>>> import numpy; print("Numpy", numpy.__version__)
Numpy 1.18.5
>>> import astropy; print("astropy", astropy.__version__)
astropy 4.0
>>> import scipy; print("Scipy", scipy.__version__)
Scipy 1.4.1
>>> import matplotlib; print("Matplotlib", matplotlib.__version__)
Matplotlib 3.0.3
```

### PR DESCRIPTION
**Pull request by aarchiba**
### Description

This pull request is to address #11500 by allowing users to select which warnings to log - all, or only subclasses of AstropyUserWarning.

Fixes #11500

### CODE REVIEW

**Comment by aarchiba** [comment]
It might be preferable to abolish the exception raising on `enable_log_warnings()`/`disable_log_warnings()` entirely so that users can simply request what they want without checking the current state.

**Comment by aarchiba** [comment]
I removed the logger's habit of raising exceptions on a double enable; I could be persuaded to put it back.

File: docs/changes/11501.feature.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+- Allow logging of all warnings. [#11500]
+
+
```
Comment (pllim) [inline docs/changes/11501.feature.rst:3]: Don't need to insert number anymore. It is captured in the filename.
```suggestion
Allow logging of all warnings.
```

**Comment by pllim** [comment]
Closing in favor of documenting #11562 . Sorry for any inconvenience caused!

**Pull request closed without merging**